internal/fwserver/server_moveresourcestate.go

```diff
@@ -221,6 +252,9 @@ func (s *Server) MoveResourceState(ctx context.Context, req *MoveResourceStateRe
 		if !moveStateResp.TargetState.Raw.Equal(tftypes.NewValue(req.TargetResourceSchema.Type().TerraformType(ctx), nil)) {
 			resp.Diagnostics = moveStateResp.Diagnostics
 			resp.TargetState = &moveStateResp.TargetState
+			if moveStateResp.TargetIdentity != nil {
+				resp.TargetIdentity = moveStateResp.TargetIdentity
+			}
```

Comment:
We can check the identity here! (you can also check it above by just using `moveStateResp.TargetIdentity`, doesn't matter which 🤷🏻 

```suggestion
			if moveStateResp.TargetIdentity != nil {
				resp.TargetIdentity = moveStateResp.TargetIdentity
			}

			if resp.TargetIdentity != nil && req.IdentitySchema == nil {
				resp.Diagnostics.AddError(
					"Unexpected Move State Response",
					"An unexpected error was encountered when creating the move state response. New identity data was returned by the provider move state operation, but the resource does not indicate identity support.\n\n"+
						"This is always a problem with the provider and should be reported to the provider developer.",
				)

				return
			}
```

Reply:
Once you update this your test should show diagnostics 👍🏻